contrib/tele.centrum.v2/index.php

```diff
@@ -3,13 +3,19 @@
 error_reporting(E_ALL &~ E_NOTICE &~ E_DEPRECATED);
 
 require_once('..' . DIRECTORY_SEPARATOR . 'initLMS.php');
+require_once('lib' . DIRECTORY_SEPARATOR . 'definitions.php');
+$basedir=(__DIR__ . DIRECTORY_SEPARATOR . 'templates_c');
 
 $uid        = $_GET['id'];
 $phone      = $_GET['phone'];
 $agentnr    = $_GET['agentnr'];
 $ticket['phonetype'] = 'on';
 
-require_once('lib' . DIRECTORY_SEPARATOR . 'definitions.php');
+if (!is_dir($basedir)) {
```

Comment:
No i widzę, że jednak uparłeś się, żeby za każdym razem sprawdzać czy katalog istnieje i czy ma odpowiednie uprawnienia... Będzie co nazwać gaszeniem pożaru...

Reply:
Skrypt ma działać z miejsca. Po co 20 osob które to wdroży ma się zastanowić co poprawić?

Reply:
Czytałeś co napisałem w innym miejscu w komentarzu odnośnie tego, że docelowo raczej trzeba się wyzbywać walidacji instalacji przy każdym uruchamianiu aplikacji www?

Reply:
Nie wiem czy w najbliższym czasie z CORE nie wyleci:
https://github.com/chilek/lms-plus/blob/20f8dbfbf7ce856d0075d6bf112ea572aaae609f/index.php#L106
Ja co do zasadności jestem przekonany, ale mam jeszcze jakieś małe obawy odnośnie tego czy ludzie będą sobie radzić nie czytając dokumentacji aktualizacyjnych (gdzie jest na tacy podane co robić).

Reply:
Zgadzam się z tym że trzeba się pozbywać walidacji. Czy smarty zwraca coś gdy nie jest w stanie zapisać pliku do tenplates_c? Może tam przesunąć to sprawdzanie...

Reply:
Jakimś pomysłem jest też dodanie do composer.json różnych komend do scripts:

np. sprawdź czy schemat db jest aktualny
np. to co sprawdza: checkdirs.php

Ewentualnie przeniesienie czynności sprawdzających tylko na stronę logowania.